core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1018 @@
+    expect(questionPlayerEngineService['questions'].length).toBe(3);
+  });
+
+  it('should record if new cards are added', () =>{
+    // eslint-disable-next-line dot-notation
+    questionPlayerEngineService['nextIndex'] = 1;
+
+    questionPlayerEngineService.recordNewCardAdded();
+
+    // eslint-disable-next-line dot-notation
+    expect(questionPlayerEngineService['currentIndex']).toBe(1);
+  });
+
+  it('should get the current question', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let currentQuestion = questionPlayerEngineService.getCurrentQuestion();
+
+    expect(currentQuestion.getId()).toBe('id1');
+  });
+
+  it('should get the current question Id', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+    let currentQuestionId = questionPlayerEngineService.getCurrentQuestionId();
+
+    expect(currentQuestionId).toBe('id1');
+  });
+
+  it('should return number of questions', () =>{
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+    let exploration = {
+      version: '1'
+    } as unknown as FetchExplorationBackendResponse;
+
+    spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    spyOn(contextService, 'getExplorationId').and.returnValue('id1');
+    spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(false);
+    spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+    spyOn(readOnlyExplorationBackendApiService, 'loadExplorationAsync')
+      .and.returnValue(Promise.resolve(exploration));
+
+    questionPlayerEngineService.init(
```

Comment:
You should typically have at least two test cases -- try loading with one backend dict, check the number of questions, try loading with a different backend dict, check that the number is now different.

Reply:
done.